### ISSUE TITLE
Add a way to trace back the reason for a node's inclusion

### ISSUE THREAD

**Issue by isaacs** [npm/arborist#115]
When we hit peer dep `ERESOLVE` errors, the data that we're sharing with the caller isn't enough to provide a reasonable explanation in many cases.  It frequently amounts to "these two things you've never heard of don't work together", which isn't helpful.

Also, it'd be nice to have a `npm explain node_modules/foo` to tell the user _why_ a given module is in the tree.  Ie, what depends on it, what depends on those things, etc., all the way back to the root node.

This can be challenging or overwhelming when a node is highly deduped.  Eg, imagine a large project with thousands of deps, and running `npm explain node_modules/mkdirp`, might end up with a ton of data being dumped.  We could still shorten the chain in the output, maybe just show some maximum number of steps and elide them with `...` unless the user sets the `--all` config switch.

The approach is basically taking the `edgesIn` links, and turning it into a tree that can be easily understood by the user.

**Comment by gajus** [comment]
How do I currently debug this?

All the error says is:

```
❯ npm i
npm ERR! code ERESOLVE
npm ERR! unable to resolve dependency tree

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/gajus/.npm/_logs/2020-08-24T21_51_31_491Z-debug.log

```

In the detail error log there is a line describing "CONFLICT":

```
1410 silly placeDep @storybook/node-logger@6.0.16 Symbol(OK) for: @storybook/addon-essentials@6.0.16
1411 silly placeDep react@^16.8.0 Symbol(CONFLICT) for: @storybook/addon-essentials@6.0.16
1412 timing idealTree Completed in 32718ms
1413 timing command:install Completed in 32721ms

```

, which I suspect is the underlying cause.

However, it is not clear how to fix this.

**Comment by isaacs** [comment]
Landed in Arborist 0.0.20, will be in next cli beta.
